_posts/2019-08-12-what-the-cluster.markdown

```diff
@@ -0,0 +1,75 @@
+---
+layout: post
+title:  "Snowflake: What the cluster"
+date:   2019-08-12
+tags: snowflake
+author: Natalie Laing
+image: img/snow.jpg
+---
+![Snowflake Unicorn](/img/blog/what-the-cluster/Uni-Snowflake.png)
+
+I recently attended a Snowflake 4 day partner bootcamp in Melbourne to get some hands on experience with Snowflake.
+
+* Snowflake is a data warehouse built for the cloud that can handle structured and semi-structured data. Snowflake supports an unlimited number of simultaneous users so multiple groups of users can access data at the same time with no performance degradation.
+* Snowflake is a zero-management data warehouse-as-a-service.
+* Snowflake deals with all the parts of setting up a data warehouse solution so you don’t have to; you get to play with all the bells and whistles to tune snowflake into your ideal data warehouse solution.
+* Snowflake deals with all the management and setting up of your data warehouse so you can focus on the processes you care about.
+* Snowflake comes with a fail safe which stores your historical data. This is non configurable and only accessible by contacting Snowflake support. Snowflake will take a copy of all tables, views, etc and store this for seven days and is only to be used to retrieve data in a case of damage or system failure. You get a seven day fail safe as default and this feature can’t be turned off.
+* You can share your tables and materialized views with consumers who do not have a snowflake account by using reader accounts.
+
+
+One of the reoccurring themes was clusters. Snowflake uses the term cluster a lot!
+
+### Data Cluster
+
+This is how data is ordered and sorted in micro-partitions. Snowflake will cluster your data automatically into micro-partitions to allow for faster retrieval of frequently requested data.Micro-partitions is Snowflakes unique way of storing large amounts of data in a way that enables fast retrieval of frequently accessed data. Micro-partitions are immutable , compressed and encrypted at rest.
+You can evaluate how your table is clustered by looking at certain columns that your users will interface with.
+
+```sql
+SELECT
+SYSTEM$CLUSTERING_INFORMATION('UNICORNS','(SIZE,AGE)');
+```
+
+### Compute Cluster
+
+A compute cluster is a collection of one or more VMs connected within a mesh. All virtual warehouses are an independent compute cluster which does not share its resources with other compute clusters. As each compute cluster does not share compute resources all other virtual warehouses will have no impact on eachother.As a standard for your warehouse you will only have a single compute cluster meaning you are unable to scale out, if you need more than one compute cluster you would need to utilise the multi cluster warehouses.
```

Comment:
`does not share compute resources all other virtual warehouses...` < need a comma after resources

Reply:
`impact on eachother.As a standard` need space

Reply:
`unable to scale out, if you need more` Make this the end of a sentence. (replace comma)

Reply:
`if you need more than one compute cluster you would need ` < needs comma after cluster